tests/e2e/specs/miscellaneous/CustomOpenVSXRegistry.spec.ts

```diff
@@ -0,0 +1,217 @@
+/** *******************************************************************
+ * copyright (c) 2023 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+import { FACTORY_TEST_CONSTANTS } from '../../constants/FACTORY_TEST_CONSTANTS';
+import { BASE_TEST_CONSTANTS } from '../../constants/BASE_TEST_CONSTANTS';
+import { BrowserTabsUtil } from '../../utils/BrowserTabsUtil';
+import { e2eContainer } from '../../configs/inversify.config';
+import { CLASSES, TYPES } from '../../configs/inversify.types';
+import { WorkspaceHandlingTests } from '../../tests-library/WorkspaceHandlingTests';
+import { ProjectAndFileTests } from '../../tests-library/ProjectAndFileTests';
+import { LoginTests } from '../../tests-library/LoginTests';
+import { ActivityBar, SideBarView, ViewItem, ViewSection } from 'monaco-page-objects';
+import { registerRunningWorkspace } from '../MochaHooks';
+import { Logger } from '../../utils/Logger';
+import { expect } from 'chai';
+import { ShellString } from 'shelljs';
+import { KubernetesCommandLineToolsExecutor } from '../../utils/KubernetesCommandLineToolsExecutor';
+import { ITestWorkspaceUtil } from '../../utils/workspace/ITestWorkspaceUtil';
+import { TIMEOUT_CONSTANTS } from '../../constants/TIMEOUT_CONSTANTS';
+import { DriverHelper } from '../../utils/DriverHelper';
+import { error } from 'selenium-webdriver';
+
+suite(
+	`Create a workspace via launching a factory from the ${FACTORY_TEST_CONSTANTS.TS_SELENIUM_FACTORY_GIT_PROVIDER} repository ${BASE_TEST_CONSTANTS.TEST_ENVIRONMENT}`,
```

Comment:
If `https://github.com/redhat-developer/devspaces` repo is mandatory, why not to use it explicitly without FACTORY_TEST_CONSTANTS.TS_SELENIUM_FACTORY_GIT_PROVIDER ?

Reply:
I can find usage of `BASE_TEST_CONSTANTS.TEST_ENVIRONMENT` in the script.
Is it really needed?

Reply:
> I can find usage of `BASE_TEST_CONSTANTS.TEST_ENVIRONMENT` in the script. Is it really needed?

removed, thx for pointing

Reply:
I see.

By the way, `TS_SELENIUM_FACTORY_GIT_REPO_URL` seems to be more relevant name to the actual value than `TS_SELENIUM_FACTORY_GIT_PROVIDER`.

Also, I would suggest setting `https://github.com/redhat-developer/devspaces` repo by default, for example: https://github.com/eclipse-che/che/blob/main/tests/e2e/specs/SmokeTest.spec.ts#L28-L29

Reply:
I removed the TS_SELENIUM_FACTORY_GIT_PROVIDER from the title last. But for me using `FACTORY_TEST_CONSTANTS.TS_SELENIUM_FACTORY_URL()`  - easier way to get a factory directly - without interactions on dashboard. che/che/blob/3ede2af2327c2bb209a9ba93d333ac274dc73547/tests/e2e/specs/miscellaneous/CustomOpenVSXRegistry.spec.ts#L53

Reply:
thanks for the fixup